vowpalwabbit/core/src/reductions/automl.cc

```diff
@@ -76,28 +76,28 @@ void persist(automl<CMType>& data, VW::metric_sink& metrics)
 }
 
 template <typename CMType>
-void finish_example(VW::workspace& all, automl<CMType>& data, VW::multi_ex& ec)
+void print_update_automl(VW::workspace& all, VW::shared_data& /* sd */, const automl<CMType>& data,
+    const VW::multi_ex& ec, VW::io::logger& /* unused */)
 {
-  interaction_vec_t* incoming_interactions = ec[0]->interactions;
+  data.adf_learner->print_example(all, ec);
+}
 
+template <typename CMType>
+void update_stats_automl(const VW::workspace& /* all */, VW::shared_data& /* sd */, const automl<CMType>& data,
+    const VW::multi_ex& ec, VW::io::logger& /* logger */)
+{
+  interaction_vec_t* incoming_interactions = ec[0]->interactions;
   uint64_t champ_live_slot = data.cm->current_champ;
   for (VW::example* ex : ec) { apply_config(ex, &data.cm->estimators[champ_live_slot].first.live_interactions); }
-
-  {
-    auto restore_guard = VW::scope_exit(
-        [&ec, &incoming_interactions]
-        {
-          for (VW::example* ex : ec) { ex->interactions = incoming_interactions; }
-        });
-
-    data.adf_learner->print_example(all, ec);
-  }
-
-  VW::finish_example(all, ec);
+  VW::scope_exit(
```

Comment:
Examples should not be mutated in this function

Reply:
none of this seems necessary here